packages/monocdk-experiment/package.json

```diff
@@ -258,14 +258,16 @@
     "@types/node": "^10.17.35",
     "cdk-build-tools": "0.0.0",
     "constructs": "^3.0.4",
+    "cdk8s": "^0.30.0",
     "fs-extra": "^9.0.1",
     "pkglint": "0.0.0",
     "ts-node": "^9.0.0",
     "typescript": "~3.8.3",
     "ubergen": "0.0.0"
   },
   "peerDependencies": {
-    "constructs": "^3.0.4"
+    "constructs": "^3.0.4",
+    "cdk8s": "^0.30.0"
```

Comment:
@iliapolo @eladb was this intentional? This means that users using monocdk will need to add to their dependencies: `monocdk`, `constructs` and `cdk8s`

Reply:
That's a very good point. Technically they will only need this if they use this feature but that's not something we can express through regular peer deps. @iliapolo Thoughts?

Reply:
Here's a trick that we might be able to use: I wonder if we should not advertise this as a peer dependency but rather just document that it's required in the closure.

Reply:
Do you mean just for monocdk? Might require some build twicks because its omission is causing a build failure. We can try it though, agree it will be a better experience.

And in case it's not going to solve this quirk for v2 where that dependency will actually be needed to compile the code...I think for v2 thats just a hit we will have to take, one of the disadvantages of having a monolithic package.

Reply:
I would try to list it as a `buildDependency` and exclude the eslint rule.

I don't think we can take the hit for monocdk. We need a solution for this type of "opt in peer". Npm has a concept of "optionalDependencies" which might be what we need

Reply:
I'll give it a try

Reply:
“I do not like it, I do not like it at all; I do not like it as a peer dependency, I do not like it as  dev dependency”

Reply:
Just to make sure we are all in agreement on the meaning of adding this as a peer-dependency (or any type of dependency which is requires the user to supply it). This means that the EKS module will always have to support the minimal version of `cdk8s` declared in monocdk, if not it will be considered a breaking change for monocdk customers.

Reply:
It's a little late now but I 100 million % agree with Neta.

It can't be an `optionalDependency`: `cdk8s.Chart` must be downloaded and exist to even COMPILE this, and to typecheck the downstream `.d.ts` files.

The only feasible solution is to use IoC and an integration package:

```
aws-eks : IKubeDefinition
aws-eks-cdk8s: class Cdk8sDefinition implements IKubeDefinition
```